src/core/client_channel/load_balanced_call_destination.cc

```diff
@@ -0,0 +1,334 @@
+                                               "LB pick");
+        }
+        // If wait_for_ready is true, then queue to retry when we get a new
+        // picker.
+        return Continue{};
+      },
+      // DropPick
+      [&](LoadBalancingPolicy::PickResult::Drop* drop_pick)
+          -> LoopCtl<absl::StatusOr<RefCountedPtr<UnstartedCallDestination>>> {
+        if (GRPC_TRACE_FLAG_ENABLED(grpc_client_channel_lb_call_trace)) {
+          gpr_log(GPR_INFO, "client_channel: %sLB pick dropped: %s",
+                  GetContext<Activity>()->DebugTag().c_str(),
+                  drop_pick->status.ToString().c_str());
+        }
+        return grpc_error_set_int(MaybeRewriteIllegalStatusCode(
+                                      std::move(drop_pick->status), "LB drop"),
+                                  StatusIntProperty::kLbPolicyDrop, 1);
+      });
+}
+
+}  // namespace
+
+void LoadBalancedCallDestination::StartCall(
+    UnstartedCallHandler unstarted_handler) {
+  // If there is a call tracer, create a call attempt tracer.
+  bool* is_transparent_retry_metadata =
+      unstarted_handler.UnprocessedClientInitialMetadata().get_pointer(
+          IsTransparentRetry());
+  bool is_transparent_retry = is_transparent_retry_metadata != nullptr
+                                  ? *is_transparent_retry_metadata
+                                  : false;
+  MaybeCreateCallAttemptTracer(is_transparent_retry);
+  // Spawn a promise to do the LB pick.
+  // This will eventually start the call.
+  unstarted_handler.SpawnGuardedUntilCallCompletes(
+      "lb_pick", [unstarted_handler, picker = picker_]() mutable {
+        return Map(
+            // Wait for the LB picker.
+            CheckDelayed(Loop(
+                [last_picker =
+                     RefCountedPtr<LoadBalancingPolicy::SubchannelPicker>(),
+                 unstarted_handler, picker]() mutable {
+                  return Map(
+                      picker.Next(last_picker),
+                      [unstarted_handler, &last_picker](
+                          RefCountedPtr<LoadBalancingPolicy::SubchannelPicker>
+                              picker) mutable {
+                        last_picker = std::move(picker);
+                        // Returns 3 possible things:
+                        // - Continue to queue the pick
+                        // - non-OK status to fail the pick
+                        // - a connected subchannel to complete the pick
+                        return PickSubchannel(*last_picker, unstarted_handler);
+                      });
+                })),
+            // Create call stack on the connected subchannel.
+            [unstarted_handler](
+                std::tuple<
+                    absl::StatusOr<RefCountedPtr<UnstartedCallDestination>>,
+                    bool>
+                    pick_result) {
+              auto& connected_subchannel = std::get<0>(pick_result);
```

Comment:
This should be called `call_destination`.

Reply:
done